src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosResponseProcessor.cs

```diff
@@ -68,9 +68,31 @@ public async Task ProcessException(Exception ex)
                 {
                     throw new Core.Exceptions.RequestEntityTooLargeException();
                 }
-                else if (dce.StatusCode == HttpStatusCode.Forbidden && dce.GetSubStatusValue() == CosmosDbSubStatusValues.CustomerManagedKeyInaccessible)
+                else if (dce.StatusCode == HttpStatusCode.Forbidden)
                 {
-                    throw new Core.Exceptions.CustomerManagedKeyInaccessibleException();
+                    switch (dce.GetSubStatusValue())
+                    {
+                        case CosmosDbSubStatusValues.CmkAadClientCredentialsGrantFailure:
```

Comment:
In commit #2 I will have one generic CMK exception, per feedback.  In this case, I think this code becomes much simpler if CosmosDbSubStatusValues is an enum.  I've done so; open to feedback.